src/tools/clippy/clippy_utils/src/macros.rs

```diff
@@ -728,7 +728,7 @@ impl<'tcx> FormatSpec<'tcx> {
         Some(Self {
             fill: spec.fill,
             align: spec.align,
-            flags: spec.flags,
+            no_flags: spec.sign.is_none() && !spec.alternate && !spec.zero_pad && spec.debug_hex.is_none(),
```

Comment:
this seems fragile, maybe add a method to `spec` that exhaustively destructures itself to make sure future field changes aren't missed by accident.

Reply:
This is just a quick patch to fix clippy for now. Clippy will be refactored pretty soon, during which this will probably disappear entirely. See https://github.com/rust-lang/rust-clippy/issues/10233